File: javascript/bubble_sort.js
window.onload = function() {
    let example = [5, 1, 6, 3, 4];
    bubble_sort(example);
    let string = "";
    for (let ind = 0; ind < example.length; ind++) {
        string += example[ind] + ' ';
    }
    console.log(string);
}

function bubble_sort(arr) {
    let l = arr.length;
    let changes = true;
    while (changes) {
        changes = false;
        for (let ind = 0; ind < l; ind++) {
            if (arr[ind] > arr[ind + 1]) {
                [arr[ind], arr[ind + 1]] = [arr[ind + 1], arr[ind]]
                changes = true;
            }
        }
    } // no need to return the array
}

File: c/bubble_sort.c
#include <stdio.h>

void bubble_sort(int arr[], int length) {
    length--;
    int changes = 1;
    while (changes) {
        changes = 0;
        for (int ind = 0; ind < length; ind++) {
            if (arr[ind] > arr[ind + 1]) {
                int swap = arr[ind];
                arr[ind] = arr[ind + 1];
                arr[ind + 1] = swap;
                changes = 1;
            }
        }
    } // no need to return the array
}

int main() {
    int example[] = {5, 1, 6, 3, 4};
    int length = sizeof(example) / 4;
    bubble_sort(example, length);
    for (int ind = 0; ind < length; ind++) {
        printf("%d ", example[ind]);
    }
    return 0;
}